string/string.h

```diff
@@ -0,0 +1,297 @@
+/**
+ * Library for embedded strings
+ *
+ * @author Aaron Chan
+ */
+
+#ifndef STRING_H
+#define STRING_H
+
+#include <cstddef>
+
+const char NULL_TERMINATOR = '\0';
+
+template<const size_t size>
+class String {
+public:
+    int strlen = 0;
+    char string[size] = {};
```

Comment:
this buffer should not be owned by this class, it should be passed from the alloc::String instead